src/routes/api.js

```diff
@@ -7,18 +7,80 @@ require('dotenv').config();
+const GOOGLE_TEXT_SEARCH_URL = 'https://maps.googleapis.com/maps/api/place/textsearch/json';
+const GOOGLE_PLACE_PHOTO = 'https://maps.googleapis.com/maps/api/place/photo';
+
+// const FOOD_URL = `https://maps.googleapis.com/maps/api/place/nearbysearch/json?location=-33.8670522,151.1957362&radius=1500&type=restaurant&keyword=cruise&key=${GOOGLE_API_KEY}`;
+// const FIND_PLACE_URL = `https://maps.googleapis.com/maps/api/place/findplacefromtext/json?input=Museum%20of%20Contemporary%20Art%20Australia&inputtype=textquery&fields=photos,formatted_address,name,rating,opening_hours,geometry&key=${GOOGLE_API_KEY}`;
+// const SEARCH_URL1 = `https://maps.googleapis.com/maps/api/place/textsearch/json?query=restaurants+in+Sydney&key=${GOOGLE_API_KEY}`;
+
+// add return if adding more functionality than just the single line
+
+const getPhotoURL = (googlePlace) => {
+  if (googlePlace.photos && googlePlace.photos.length && googlePlace.photos[0].photo_reference) {
+    return `${GOOGLE_PLACE_PHOTO}?maxwidth=400&photoreference=${googlePlace.photos[0].photo_reference}&key=${GOOGLE_API_KEY}`;
+  }
+  return 'https://via.placeholder.com/400';
+};
+
+const transformGooglePlaces = (googlePlaces = []) => googlePlaces.map((googlePlace) => ({
+  name: googlePlace.name,
+  address: googlePlace.formatted_address,
+  placeId: googlePlace.place_id,
+  price: googlePlace.price_level,
+  rating: googlePlace.rating,
+  photo: getPhotoURL(googlePlace),
+  type: googlePlace.types,
+}));
 
 const getPlacesFromGoogle = async (req, res) => {
   try {
-    // const { searchTerm } = req.body;
-
-    const { data } = await axios.get(SEARCH_URL);
-    // const IMAGE_URL = `https://maps.googleapis.com/maps/api/place/photo?maxwidth=400&photoreference=${data.results.photos[0].photo_reference}&key=${GOOGLE_API_KEY}`;
-    console.log(data, FOOD_URL, FIND_PLACE_URL);
-    res.json(data);
+    const {
+      searchTerm, food = false, nightlife = false, activities = false,
+    } = req.body;
+
+    let foodResults = [];
+    let nightlifeResults = [];
+    let activitiesResults = [];
+
+    if (food) {
+      const { data: foodData } = await axios.get(
+        GOOGLE_TEXT_SEARCH_URL,
+        {
+          params: {
+            query: `food+in+${searchTerm}`,
+            key: GOOGLE_API_KEY,
+          },
+        },
+      );
+      foodResults = transformGooglePlaces(foodData.results);
+    }
+    if (nightlife) {
+      const { data: nightlifeData } = await axios.get(
+        GOOGLE_TEXT_SEARCH_URL,
+        {
+          params: {
+            query: `nightlife+in+${searchTerm}`,
+            key: GOOGLE_API_KEY,
+          },
+        },
+      );
+      nightlifeResults = transformGooglePlaces(nightlifeData.results);
+    }
+    if (activities) {
+      const { activitiesData } = await axios.get(
+        GOOGLE_TEXT_SEARCH_URL,
+        {
+          params: {
```

Comment:
Could maybe move all the params stuff in a function `getParams('food')` which returns the expected params object